pyrestriction/model.py

```diff
@@ -123,10 +132,22 @@ def amount(self):
     def debt(self):
         return self._debt
 
+    def __repr__(self):
+        constructor_parameters = list(signature(self.__class__.__init__).parameters)
+        constructor_parameters.pop(0)
+        textual_argument = ["{p} = {a}".format(p = parameter, a = self.__getattribute__("_"+parameter))
+                            for parameter in constructor_parameters]
+        arguments = ", ".join(textual_argument)
+
+        classname = self.__class__.__name__
+
+        return "{classname}({arguments})".format(classname=classname, arguments=arguments)
+
 class SavingOperation(Operation):
     """Save money during one period"""
     def __init__(self, amount):
         super(SavingOperation, self).__init__(amount)
+        self._amount = amount
```

Comment:
This smells a bit. Why do you have the same private variable in the superclass and the subclass?